### PR TITLE
MacOS support

### PR DESCRIPTION
**Pull request by jshah4517**
The setup script didn't allow use of MacOS, even though it's possible to install using Docker. This PR adds support for running on MacOS, and also fixes an issue with the `sed` commands where `-E` is not available on MacOS.

### CODE REVIEW

File: templates/docker-compose/setup.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -eu -o pipefail
 
-usage="Usage (Linux): curl -LsS https://raw.githubusercontent.com/supportpal/helpdesk-install/master/templates/docker-compose/setup.sh | bash
+usage="Usage (Linux / MacOS): curl -LsS https://raw.githubusercontent.com/supportpal/helpdesk-install/master/templates/docker-compose/setup.sh | bash
```
Comment (bytestream) [inline templates/docker-compose/setup.sh:4]:
```suggestion
usage="Usage (Linux / MacOS): bash <(curl -LsS https://raw.githubusercontent.com/supportpal/helpdesk-install/master/templates/docker-compose/setup.sh)
```

This line should match https://docs.supportpal.com/current/Deploy+on+Docker

Does it still work on MacOS?

File: templates/docker-compose/setup.sh [hunk 2]
```diff
@@ -208,15 +212,15 @@ configure() {
   fi
 
   if [[ -n "${email// }" ]]; then
-    sed -i -E "s/^(MAILTO=).*/\1$(escape_re "${email// }")/" .env
+    sed -i "" -e "s/^\(MAILTO=\).*/\1$(escape_re "${email// }")/" .env
```
Comment (bytestream) [inline templates/docker-compose/setup.sh:215]: `sed -i ""` seems to be unique to sed posix / macos

we'll need to do https://stackoverflow.com/a/57766728

File: templates/docker-compose/setup.sh (hunk 1 above)
Comment (jshah4517) [reply templates/docker-compose/setup.sh]: Yes, I've been using the command from the guide.

**Comment by bytestream** [comment]
LGTM :+1:

**Pull request merged**